pcap/search.go

```diff
@@ -0,0 +1,172 @@
+package pcap
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+var (
+	// ErrNoPacketsFound is an error indicating no packets have been found.
+	ErrNoPacketsFound = errors.New("no packets found")
+)
+
+//XXX bool true = TCP, else UDP, change this
+type Predicate func(bool, Socket, Socket) bool
+
+// Search describes the parameters for a packet search over a pcap file.
+type Search struct {
+	span    nano.Span
+	matcher Predicate
+	id      string
+}
+
+// NewSearch creates a new search object.
+func NewSearch(span nano.Span, proto string, srcHost string, srcPort *uint16, dstHost string, dstPort *uint16) (*Search, error) {
+	switch proto {
+	//case "icmp", "tcp", "udp":
```

Comment:
drop this?

Reply:
It's a reminder that we need to add support for icmp and other filter schemes.  I updated the comment.